### ISSUE TITLE
Deleting container images is non-deterministic

### ISSUE THREAD

**Issue by delvh** [go-gitea/gitea#25559]
### Description

At the moment, it is only somewhat possible to delete `container` packages:
When you come from the org `packages` tab and click on `delete` for any package, you simply get redirected to the org page listing all packages.
When you come from the repo `packages` tab, there is a chance of about 50-60% that you will get the same behavior as with the org `packages` tab.
However, after trying about three to four times per package, I was finally able to delete one.

### Gitea Version

1.19.3

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

**Nothing**. I mean, we only set the log level to `WARNING`, but apart from a lot of (somewhat) unrelated `2023/06/28 11:23:44 ...les/cache/context.go:71:SetContextData() [W] [649bfc20] cannot get cache context when setting data: *db.Context.WithValue(type struct {}, val NotifyPackageDelete)` messages, there was nothing.

### How are you running Gitea?

From binary.

### Database

SQLite

**Comment by KN4CK3R** [comment]
There may be a misunderstanding. You only delete a package **version** from the ui. If there are multiple versions the package is still there and you can view it from the package list. It may be better to redirect to the package again if there is still another version than the redirect to the package list after the delete.

**Comment by delvh** [comment]
Okay, if that's the case, then we should probably indicate that you only delete **the latest version** by "deleting the package".
Or alternatively add a `delete all versions` button.

**Comment by KN4CK3R** [comment]
No, not the latest version. You delete the version you are looking at. The version is mentioned in popup:
![grafik](https://github.com/go-gitea/gitea/assets/1666336/9db60d31-a7fc-48fa-a869-ea667954a37f)

But yes, this area should mention "version" somewhere:
![grafik](https://github.com/go-gitea/gitea/assets/1666336/beed2e5c-c514-47bf-bd70-b7126d30a142)

**Comment by silverwind** [comment]
Container version deletion now consistently removes the selected tag, preserves other tags and redirects correctly when only internal package data remains. The UI also distinguishes version deletion from package deletion.

Fixed in https://github.com/go-gitea/gitea/commit/cf2356062fd13c51c9ccc86df96e0ac6cd764f28, https://github.com/go-gitea/gitea/commit/7480aacdad9470458fa0142beab54b8011bbac25, https://github.com/go-gitea/gitea/commit/ca51b4f875276b9bae8816d550e60e2657c96056.

Written by Codex.
